lib/hashes.md

```diff
@@ -5,40 +5,42 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+the list.  Hashes can help group different items/objects in the list into associations.  Like what I know from python, the key is the sorting, association-making variables that help group, label, and identify the values assigned to them.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Since hashes do not store the items in a numbered list, so if you need the data stored in a specific order, so that it can be called by referencing a specific number in that sequence, then hashes can be useful.  They can group and organize data/objects,
+and make it so that items can be called without havinig to memorize their specific number in the ordered list or array.  Arrays store named data, so the use of the name can do a lot of things with the data associated with that key.  A type of data that can 
+benefit from the use of hashes is counting data, as it can store amounts of a certain type of item (as opposed to arrays).
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
-
+Some of this response could be found in the previous answer above, but I'll try to repeat as much here.  Both arrays and hashes are collections of information, stored in a form of repository.  The differences come mainly from how that data is stored, and how that data can be used or called.
+Arrays store the data/objects in a specific order, and particular items can be called by utilizing the number of the item in that list (beginning with zero).  Hashes have keys, which are unique names that can be used to identify values.  The keys can be used to help do a variery of things, like 
+organize a count of like items, or categorize items into desired groups, etc.  Hashes have much more variety in terms of what they can offer.
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
-
+A particular value from a hash is retrieved by using the key for that value.  For instance, if this is our hash:  grades = { "Mike Ducey" => A, "Pat Ducey" => B+ }, you reference the hash like an array, but instead of specifying the number of the item, you use the key.  So, in this case, if I write grades["Mike Ducey"], I shall get => A as a response.
 # How do you add information to a Hash?
 
-Replace me with your response...
-
+This is done by using the "bracket-equals method" (https://github.com/learn-co-curriculum/hash-overview-readme).  That is, if you have a hash and you want another key in the hash, you can write the name of the key in the brackets and the value after the equal sign.  I'll add a grade to my grades hash: grades["Kevin Ducey"] = A-
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This is achieved by using the each function.  If you want to change each value, you use something of the form: grades.each do |x| where x is the change to be done using code.
```

Comment:
Here, `x` actually represents a block parameter -- a parameter that would be passed to some block of code inside of your `each` block. That internal block is the operation you wish to perform. 

In your example, you're passing one parameter `x` to the presumed block that would come after what you have typed. 

The `each` method _can_ work with one block parameter. But now, that `x` represents _both_ the key and the value for each key-value pair.

Try running the following in a Ruby compiler ([Repl.it](https://repl.it/languages/ruby) or [Ideone](http://ideone.com/)):

```ruby
grades = {"Alex" => 90, "Sumeet" => 98, "Khalil" => 85}
grades.each do |x|
  puts x
  puts " " 
end #don't forget to end out your each block
```

How might you alter the code above to instead use separate parameters for the key and value in each key-value pair?  [Ruby documentation](http://ruby-doc.org/core-1.8.7/Hash.html#method-i-each) might help.